.github/workflows/run-unit-tests.yml

```diff
@@ -0,0 +1,66 @@
+name: Run unit tests
+
+on:
+  pull_request:
+    types: [opened,reopened,synchronize]
+
+jobs:
+  test-unit-jest:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: 'lts/*'
```

Comment:
we probably need to pin the version, as not doing so will result in non-deterministic tests.

Reply:
I copied this from the other workflows. Every workflow seems to be using this for some reason.

Reply:
Ah, the GitHub workflows haven't received much scrutiny over the years. As we move more and more things over to GitHub I think we'll find more improvements like this.

Reply:
Oh actually we can bind the node-version to the -nvmrc, so I'll just keep that.

node-version-file: '.nvmrc'